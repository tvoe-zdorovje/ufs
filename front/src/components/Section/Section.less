@import "../Shared";

.section {
  margin-top: 12px;
  border: 1px solid @green;
}

.header {
  padding: 1px 8px 1px 4px;
  margin: -12px 0 0 8px;
  color: @green;
  background-color: #fff;
  position: absolute;
  z-index: 10;
}

.container {
  padding: 10px 5px 5px 5px;
}
