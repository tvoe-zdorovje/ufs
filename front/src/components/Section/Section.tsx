import React from 'react'

const styles = require('./Section.less')

interface SectionProps {
  header: string,
  children?: any,
}

export default (props: SectionProps) => (
  <section className={styles.section}>
    <div className={styles.header}>{props.header}</div>
    <div className={styles.container}>
      {props.children}
    </div>
  </section>
);
