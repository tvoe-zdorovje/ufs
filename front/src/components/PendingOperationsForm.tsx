import React from 'react'
import {
  Layout,
  Row,
  Col,
  Table,
  AutoComplete,
  Input,
  Icon,
  Button,
  Modal
} from 'antd'
import Section from './Section/Section'

import OperationSearchForm from 'components/OperationSearchForm'
import { OVN } from 'interfaces/dto/OVN.interface'

const styles = require('./Shared.less')

const columns = [
  {
    title: 'ID пакета',
    dataIndex: 'packageId',
  },
  {
    title: 'ID задачи',
    dataIndex: 'taskId',
  },
  {
    title: 'Клиент',
    dataIndex: 'legalEntityShortName',
  },
  {
    title: 'ФИО Клиента/Представителя',
    dataIndex: 'repFio',
  },
  {
    title: 'Дата/время',
    dataIndex: 'createdDate',
  },
  {
    title: 'Номер счета',
    dataIndex: 'accountId',
  },
  {
    title: 'Сумма',
    dataIndex: 'amount',
  },
  {
    title: 'Тип операции',
    dataIndex: 'operationType',
    render: () => 'Взнос наличных на счет корпоративной/бюджетной карты',
  }
]

export interface PendingOperationsFormProps {
  load: Function,
  loadPendingOperations: Function,
  nextStepOvn: Function,
  filter: Function,
  pendingOperations: {
    items: Array<any>,
    seed: Array<any>,
  }
}

export default class PendingOperationsForm extends React.Component<PendingOperationsFormProps, any> {
  state = { showSearchByClientForm: false, selectedRows: [] }

  handleCancel = () => {
    this.setState({
      ...this.state,
      showSearchByClientForm: false,
    })
  }

  toggleModal = () => {
    this.setState({
      ...this.state,
      showSearchByClientForm: !this.state.showSearchByClientForm,
    })
  }

  rowSelection = {
    type: ('radio' as any),
    onChange: (key: string[], selectedRows: Array<OVN>) => {
      this.setState({
        ...this.state,
        selectedRows
      })
    },
  }

  nextStepOvn = () => {
    this.props.nextStepOvn(this.state.selectedRows[0])
  }

  componentDidMount() {
    const { load, loadPendingOperations } = this.props
    load()
    loadPendingOperations()
  }

  render() {
    const rowSelected = this.state && this.state.selectedRows && this.state.selectedRows.length > 0

    const {
      filter,
      pendingOperations: { seed },
    } = this.props

    const autocompletePackageIdItems =
      [...new Set(seed.map(item => item.packageId))]

    const autocompleteTaskIdItems =
      [...new Set(seed.map(item => item.taskId))]

    return (
      <Layout style={{ marginTop: 12 }}>
        <Row style={{ marginBottom: 8 }}>
          <Col span={24}>
            <AutoComplete
              className={styles.autocomplete}
              optionLabelProp="value"
              allowClear
              key="packageId"
              transitionName=""
              dataSource={(autocompletePackageIdItems.map((item, i) => (
                  <AutoComplete.Option key={i} value={item}>
                    {item}
                  </AutoComplete.Option>
                )
              ) as any)}
              onSelect={(packageId) => {
                filter({ packageId })
              }}
              onChange={(packageId) => {
                if (!packageId) {
                  filter({ packageId: false })
                }
              }}
            >
              <Input
                placeholder="Id пакета..."
                prefix={
                  <Icon type="search" className={styles.icon} />
                }
              />
            </AutoComplete>
            <AutoComplete
              className={styles.autocomplete}
              optionLabelProp="value"
              allowClear
              key="taskId"
              transitionName=""
              dataSource={(autocompleteTaskIdItems.map((item, i) => (
                  <AutoComplete.Option key={i} value={item}>
                    {item}
                  </AutoComplete.Option>
                )
              ) as any)}
              onSelect={(taskId) => {
                filter({ taskId })
              }}
              onChange={(taskId) => {
                if (!taskId) {
                  filter({ taskId: false })
                }
              }}
            >
              <Input
                placeholder="Id операции..."
                prefix={
                  <Icon type="search" className={styles.icon} />
                }
              />
            </AutoComplete>
            <Button style={{ float: 'right' }} type="primary" onClick={this.toggleModal}>
              Поиск по клиенту/представителю
            </Button>
          </Col>
        </Row>
        <Row>
          <Section header="Незавершенные операции">
            <Modal
              footer={[
                <Button key="submit" type="primary" size="large">
                  Найти
                </Button>
              ]}
              title="Поиск операции по клиенту/представителю"
              visible={this.state.showSearchByClientForm}
              onCancel={this.handleCancel}
            >
              <OperationSearchForm />
            </Modal>
            <Table
              rowKey="taskId"
              rowSelection={this.rowSelection}
              columns={columns}
              dataSource={this.props.pendingOperations.items}
              locale={{
                filterConfirm: 'ОК',
                filterReset: 'Сбросить',
                emptyText: 'Данные отсутствуют',
              }}
            />
          </Section>
        </Row>
        <Row className={styles.rightAligned}>
          <Button className={rowSelected ? styles.greenButton : ''} disabled={ !rowSelected }
                  onClick={this.nextStepOvn}>
            Открыть операцию
          </Button>
        </Row>
      </Layout>
    )
  }
}
