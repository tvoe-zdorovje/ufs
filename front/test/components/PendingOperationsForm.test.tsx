import React from 'react'
import {
  shallow,
  render
} from 'enzyme'
import toJson from 'enzyme-to-json'
import {
  Table,
  AutoComplete,
} from 'antd'

import PendingOperationsForm, { PendingOperationsFormProps } from 'components/PendingOperationsForm'
import Section from 'components/Section/Section'

describe('Компонент PendingOperationsForm', () => {
  const items = [{
    id: '1',
    packageId: '89731236',
    incomingDate: 'dd.mm.yyyy hh24:mi:ss',
    credentials: 'Петров Петр Петрович',
    clientType: 'ЮЛ',
    operationType: 'Взнос наличных на счет корпоративной/бюджетной карты',
  }, {
    id: '2',
    packageId: '89731238',
    incomingDate: 'dd.mm.yyyy hh24:mi:ss',
    credentials: 'Петров Петр Петрович',
    clientType: 'ЮЛ',
    operationType: 'Взнос наличных на счет корпоративной/бюджетной карты',
  },]

  const props: PendingOperationsFormProps = {
    load: () => {
    },
    loadPendingOperations: () => {
    },
    nextStepOvn: () => {
    },
    filter: () => true,
    pendingOperations: {
      loaded: true,
      items: items,
      seed: items,
    }
  }

  const shallowWrapper = shallow(<PendingOperationsForm {...props} />)
  const renderWrapper = render(<PendingOperationsForm {...props} />)

  it('отображает компонент <Table/>', () => {
    expect(shallowWrapper.find(Table)).toHaveLength(1)
  })

  it('отображает компонент <Section/>', () => {
    expect(shallowWrapper.find(Section)).toHaveLength(1)
  })

  it('отображает компонент <AutoComplete/>', () => {
    expect(shallowWrapper.find(AutoComplete)).toHaveLength(2)
  })

  it('совпадает со своим снимком', () => {
    expect(toJson(renderWrapper)).toMatchSnapshot()
  })
})
