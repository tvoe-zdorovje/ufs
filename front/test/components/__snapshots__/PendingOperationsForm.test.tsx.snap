// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент PendingOperationsForm совпадает со своим снимком 1`] = `
<div
  class="ant-layout"
  style="margin-top:12px;"
>
  <div
    class="ant-row"
    style="margin-bottom:8px;"
  >
    <div
      class="ant-col-24"
    >
      <div
        class="autocomplete ant-select-show-search ant-select-auto-complete ant-select ant-select-combobox ant-select-enabled ant-select-allow-clear"
      >
        <div
          aria-autocomplete="list"
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-select-selection
                    ant-select-selection--single"
          role="combobox"
        >
          <div
            class="ant-select-selection__rendered"
          >
            <ul>
              <li
                class="ant-select-search ant-select-search--inline"
              >
                <div
                  class="ant-select-search__field__wrap"
                >
                  <span
                    class="ant-input-affix-wrapper"
                  >
                    <span
                      class="ant-input-prefix"
                    >
                      <i
                        class="anticon anticon-search icon"
                      />
                    </span>
                    <input
                      class="ant-input ant-input ant-select-search__field"
                      placeholder="Id пакета..."
                      type="text"
                      value=""
                    />
                  </span>
                  <span
                    class="ant-select-search__field__mirror"
                  >
                     
                  </span>
                </div>
              </li>
            </ul>
          </div>
          <span
            class="ant-select-selection__clear"
            style="user-select:none;-webkit-user-select:none;display:none;"
            unselectable="unselectable"
          />
          <span
            class="ant-select-arrow"
            style="user-select:none;-webkit-user-select:none;"
            unselectable="unselectable"
          >
            <b />
          </span>
        </div>
      </div>
      <div
        class="autocomplete ant-select-show-search ant-select-auto-complete ant-select ant-select-combobox ant-select-enabled ant-select-allow-clear"
      >
        <div
          aria-autocomplete="list"
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-select-selection
                    ant-select-selection--single"
          role="combobox"
        >
          <div
            class="ant-select-selection__rendered"
          >
            <ul>
              <li
                class="ant-select-search ant-select-search--inline"
              >
                <div
                  class="ant-select-search__field__wrap"
                >
                  <span
                    class="ant-input-affix-wrapper"
                  >
                    <span
                      class="ant-input-prefix"
                    >
                      <i
                        class="anticon anticon-search icon"
                      />
                    </span>
                    <input
                      class="ant-input ant-input ant-select-search__field"
                      placeholder="Id операции..."
                      type="text"
                      value=""
                    />
                  </span>
                  <span
                    class="ant-select-search__field__mirror"
                  >
                     
                  </span>
                </div>
              </li>
            </ul>
          </div>
          <span
            class="ant-select-selection__clear"
            style="user-select:none;-webkit-user-select:none;display:none;"
            unselectable="unselectable"
          />
          <span
            class="ant-select-arrow"
            style="user-select:none;-webkit-user-select:none;"
            unselectable="unselectable"
          >
            <b />
          </span>
        </div>
      </div>
      <button
        class="ant-btn ant-btn-primary"
        style="float:right;"
        type="button"
      >
        <span>
          Поиск по клиенту/представителю
        </span>
      </button>
    </div>
  </div>
  <div
    class="ant-row"
  >
    <section
      class="section"
    >
      <div
        class="header"
      >
        Незавершенные операции
      </div>
      <div
        class="container"
      >
        <div
          class="ant-table-wrapper"
        >
          <div
            class="ant-spin-nested-loading"
          >
            <div
              class="ant-spin-container"
            >
              <div
                class="ant-table ant-table-large ant-table-scroll-position-left"
              >
                <div
                  class="ant-table-content"
                >
                  <div
                    class="ant-table-body"
                  >
                    <table
                      class=""
                    >
                      <colgroup>
                        <col />
                        <col />
                        <col />
                        <col />
                        <col />
                        <col />
                        <col />
                        <col />
                        <col />
                      </colgroup>
                      <thead
                        class="ant-table-thead"
                      >
                        <tr>
                          <th
                            class="ant-table-selection-column"
                          >
                            <span />
                          </th>
                          <th
                            class=""
                          >
                            <span>
                              ID пакета
                            </span>
                          </th>
                          <th
                            class=""
                          >
                            <span>
                              ID задачи
                            </span>
                          </th>
                          <th
                            class=""
                          >
                            <span>
                              Клиент
                            </span>
                          </th>
                          <th
                            class=""
                          >
                            <span>
                              ФИО Клиента/Представителя
                            </span>
                          </th>
                          <th
                            class=""
                          >
                            <span>
                              Дата/время
                            </span>
                          </th>
                          <th
                            class=""
                          >
                            <span>
                              Номер счета
                            </span>
                          </th>
                          <th
                            class=""
                          >
                            <span>
                              Сумма
                            </span>
                          </th>
                          <th
                            class=""
                          >
                            <span>
                              Тип операции
                            </span>
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="ant-table-tbody"
                      >
                        <tr
                          class="ant-table-row  ant-table-row-level-0"
                        >
                          <td
                            class="ant-table-selection-column"
                          >
                            <span>
                              <label
                                class="ant-radio-wrapper"
                              >
                                <span
                                  class="ant-radio"
                                >
                                  <input
                                    class="ant-radio-input"
                                    type="radio"
                                  />
                                  <span
                                    class="ant-radio-inner"
                                  />
                                </span>
                              </label>
                            </span>
                          </td>
                          <td
                            class=""
                          >
                            <span
                              class="ant-table-row-indent indent-level-0"
                              style="padding-left:0px;"
                            />
                          </td>
                          <td
                            class=""
                          >
                            1
                          </td>
                          <td
                            class=""
                          />
                          <td
                            class=""
                          />
                          <td
                            class=""
                          />
                          <td
                            class=""
                          />
                          <td
                            class=""
                          />
                          <td
                            class=""
                          >
                            Взнос наличных на счет корпоративной/бюджетной карты
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row  ant-table-row-level-0"
                        >
                          <td
                            class="ant-table-selection-column"
                          >
                            <span>
                              <label
                                class="ant-radio-wrapper"
                              >
                                <span
                                  class="ant-radio"
                                >
                                  <input
                                    class="ant-radio-input"
                                    type="radio"
                                  />
                                  <span
                                    class="ant-radio-inner"
                                  />
                                </span>
                              </label>
                            </span>
                          </td>
                          <td
                            class=""
                          >
                            <span
                              class="ant-table-row-indent indent-level-0"
                              style="padding-left:0px;"
                            />
                          </td>
                          <td
                            class=""
                          >
                            2
                          </td>
                          <td
                            class=""
                          />
                          <td
                            class=""
                          />
                          <td
                            class=""
                          />
                          <td
                            class=""
                          />
                          <td
                            class=""
                          />
                          <td
                            class=""
                          >
                            Взнос наличных на счет корпоративной/бюджетной карты
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
              <ul
                class="ant-pagination ant-table-pagination"
                unselectable="unselectable"
              >
                <li
                  aria-disabled="true"
                  class="ant-pagination-disabled ant-pagination-prev"
                  title="上一页"
                >
                  <a />
                </li>
                <li
                  class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                  title="1"
                >
                  <a>
                    1
                  </a>
                </li>
                <li
                  aria-disabled="true"
                  class="ant-pagination-disabled ant-pagination-next"
                  title="下一页"
                >
                  <a />
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
  <div
    class="ant-row rightAligned"
  >
    <button
      class="ant-btn"
      disabled=""
      type="button"
    >
      <span>
        Открыть операцию
      </span>
    </button>
  </div>
</div>
`;
